src/kit.ts

```diff
@@ -798,17 +800,31 @@ async function scanDirForClangForMSVCKits(dir: string, vsInstalls: VSInstallatio
             const vs_arch = (version.target && version.target.triple.includes('i686-pc')) ? 'x86' : 'amd64';
 
             const clangArch = (vs_arch === "amd64") ? "x64\\" : "";
-            const clangKitName = `Clang ${version.version} ${clang_cli} for MSVC ${vs.installationVersion} (${install_name} - ${vs_arch})`;
-            if (binPath.startsWith(`${vs.installationPath}\\VC\\Tools\\Llvm\\${clangArch}bin`) && await util.checkFileExists(util.lightNormalizePath(binPath))) {
-                clangKits.push({
-                    name: clangKitName,
-                    visualStudio: kitVSName(vs),
-                    visualStudioArchitecture: vs_arch,
-                    compilers: {
-                        C: binPath,
-                        CXX: binPath
-                    }
-                });
+            const useClangVersion: boolean = versionLess(version.version, '14.0.0');
+            const clangKitName: string = (isClangCL ? `Clang-cl` : `Clang`) + (useClangVersion ? ` ${version.version}` : "") + ` ${clang_cli} for ${install_name} - ${vs_arch} (MSVC ${vs.installationVersion})`;
+            const clangCXX = isClangCL ? binPath : binPath.replace(/^clang/, 'clang++');
+            const installationVersion = /^(\d+)+./.exec(vs.installationVersion);
+            let generatorName: string | undefined;
+            if (installationVersion) {
+                generatorName = VsGenerators[installationVersion[1]];
```

Comment:
Let's declare the variable here and then move the check on line 811 inside this block.  You can then remove the check from line 812 and just push the kit.